library/L1_Peripheral/lpc17xx/system_controller.hpp

```diff
@@ -117,273 +53,563 @@ class SystemController final : public sjsu::SystemController
+    kMain = 0b01,
+    /// The RTC oscillator which has a frequency of approximately 32kHz.
+    kRtc = 0b10,
+  };
+
+  /// The available CPU clock sources.
+  enum class CpuClockSource : uint8_t
+  {
+    /// Uses the system clock.
+    kSystemClock = 0b0,
+    /// Uses the PLL0 clock.
+    kPll0 = 0b1,
+  };
+
+  /// The available USB subsystem clock sources.
+  ///
+  /// @warning PLL0 should not be used as a source if PLL0 is driven by the
+  ///          internal oscillator as the clock signal will not be precise
+  ///          enough for the USB subsystem.
+  enum class UsbClockSource : uint8_t
+  {
+    //! @cond Doxygen_Suppress
+    kPll0 = 0b0,
+    kPll1 = 0b1,
+    //! @endcond
+  };
+
+  /// The available USB clock dividers that can be selected when the USB clock
+  /// is driven by PLL0. Since PLL0 has an output between 275 MHz to 550 MHz,
+  /// only dividers values of 6, 8, and 10 are available in order to obtain a
+  /// 48 MHz USB clock.
+  enum class UsbClockDivider : uint8_t
   {
-    /// IRC, main, or RTC oscillator select bits
-    static constexpr bit::Mask kSelect = bit::CreateMaskFromRange(0, 1);
+    /// The divider to use when PLL0 is configured to output 288 MHz.
+    kDivideBy6 = 0b0101,
+    /// The divider to use when PLL0 is configured to output 384 MHz.
+    kDivideBy8 = 0b0111,
+    /// The divider to use when PLL0 is configured to output 480 MHz.
+    kDivideBy10 = 0b1001,
   };
 
-  /// Namespace of Clock register bitmasks
-  struct CpuClock  // NOLINT
+  /// The available peripheral clock dividers.
+  ///
+  /// @see Table 42. Peripheral Clock Selection register bit values
+  ///      https://www.nxp.com/docs/en/user-guide/UM10360.pdf#page=59
+  enum class PeripheralClockDivider : uint8_t
   {
-    /// CPU clock divider amount bitmask
-    static constexpr bit::Mask kDivider = bit::CreateMaskFromRange(0, 7);
+    //! @cond Doxygen_Suppress
+    kDivideBy4 = 0b00,
+    kDivideBy1 = 0b01,
+    kDivideBy2 = 0b10,
+    //! @endcond
+
+    /// @warning This divider value is only available for CAN peripherals.
+    kDivideBy6 = 0b11,
+    /// @warning This divider value is only available for non-CAN peripherals.
+    kDivideBy8 = 0b100,
   };
 
-  /// Bit masks for the Main PLL register
-  struct MainPll  // NOLINT
+  /// Namespace for the Clock Source Select register (CLKSRCSEL) bit masks. The
+  /// CLKSRCSEL register is used to select the oscillator used to drive the
+  /// system clock and PLL0.
+  struct ClockSourceSelectRegister  // NOLINT
   {
-    // The following bit masks apply to the PLL0STAT and PLL0CFG registers
-    /// PLL multiplier mask
-    static constexpr bit::Mask kMultiplier = bit::CreateMaskFromRange(0, 14);
-    /// PLL Pre-divider mask
-    static constexpr bit::Mask kPreDivider = bit::CreateMaskFromRange(16, 23);
+    /// Clock source select bit mask.
+    static constexpr bit::Mask kSelectMask = bit::CreateMaskFromRange(0, 1);
```

Comment:
```suggestion
    static constexpr auto kSelectMask = bit::CreateMaskFromRange(0, 1);
```
Should use auto to keep consistent across the other structs.